packages-content-model/roosterjs-content-model-editor/lib/modelApi/table/applyTableFormat.ts

```diff
@@ -202,47 +226,54 @@ function formatBackgroundColors(
 
                 setTableCellBackgroundColor(cell, color);
             }
+
+            // Format Vertical Align
+            if (!metaOverrides.vAlignOverrides[rowIndex][colIndex]) {
+                cell.format.verticalAlign = 'top';
```

Comment:
Instead of hard code "top" here, can we put it as part of the table metadata, then read this value from table metadata?

Otherwise, if we reformat an existing table which has valign to middle, all cells will suddenly align to top since they don't have cell override.

The align to top behavior should only affect new table, but for existing tables, the should keep no change.

Reply:
How is Shading handled in that case? Because Shading would have the same problem in that case since it is not part of the table Metadata.

Reply:
Shading is defined as part of the format metadata, and we read it from line 218:

<img width="474" alt="image" src="https://github.com/microsoft/roosterjs/assets/23065085/51e6ff89-a8af-41b7-957c-89b3c04babd7">

Reply:
I feel that if we put this setting inside the format metadata, we should only set its value when insert new table. For other cases that format an existing table, do not touch this value and keep whatever value it has. That means:
1. For new table, we apply a format that has all other existing values (borders, ...) as well as this vertical alignment
2. When apply format to an existing table, no matter if it already has format or vertical align, do not change its vertical align value, keep whatever it has
3. When apply the format, read this value from table, and read if it has override from cell. If it has value from table but not overridden from cell, apply this value.

Reply:
I think the way shading is done is misnamed. If it is supposed to be the default shading it should say it somewhere, but the name implies it will only be used on even rows. Should I do a name change, or just add comments?

As for new tables, applyTableFormat has no way to differentiate if the table is new or existing. So it will override existing format as it currently does for shading in content model tables. For example, if a tables' 'bgColorEven' metadata is different from null, when applyTableFormat runs it will change it to null, unless it had an override.

Reply:
The name is a little confusing. But for compatibility, let's just keep it. Feel free to add comment.

applyTableFormat doesn't need to know if a table is new. We just need to let insertTable API build a correct table format and pass into applyTableFormat.

For example, for a new table, insertTable() can add something like "verticalAlign: top" into the table format, then call applyTableFormat(). For existing table, just read the table format from the table, then combine with the new format. If the table format already has verticalAlign, it will be preserved. If not, we don't add it.